dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/ArtifactProblem.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.opensource.dependencies;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.graph.DependencyNode;
+
+/** Problem in a Maven artifact. */
+public final class ArtifactProblem {
+
+  /** The Maven artifact that has the problem. */
+  private final Artifact artifact;
+
+  /** The type of the problem in the artifact. */
+  private final Type type;
```

Comment:
Should this be a subclass instead? That is, should there be subclasses for different types of artifact problems? Using a type object feels not very object oriented.